File: mongo-service/models/Comment.js
const mongoose = require('mongoose');

const commentSchema = new mongoose.Schema({
  issueId: { type: Number, required: true, index: true },
  userId: { type: Number, required: true, index: true },
  content: { type: String, required: true },
  createdAt: { type: Date, default: Date.now },
  parentId: { type: mongoose.Schema.Types.ObjectId, index: true }
});

const Comment = mongoose.model('Comment', commentSchema);

module.exports = Comment;

File: issue-service/src/main/java/com/incidenciasvlc/issueservice/model/Comment.java
package com.incidenciasvlc.issueservice.model;

import java.util.Date;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Comment {
    private String _id;
    private Integer issueId;
    private Integer userId;
    private String content;
    private Date createdAt;
    private String parentId;

    private User user;
}
